internal/integration/unified/crud_helpers.go

```diff
@@ -63,6 +62,70 @@ func createUpdateArguments(args bson.Raw) (*updateArguments, error) {
 		case "let":
 			ua.opts.SetLet(val.Document())
 		case "update":
+			var err error
+			ua.update, err = createUpdateValue(val)
+			if err != nil {
+				return nil, fmt.Errorf("error processing update value: %w", err)
+			}
+		case "upsert":
+			ua.opts.SetUpsert(val.Boolean())
+		default:
+			return nil, fmt.Errorf("unrecognized update option %q", key)
+		}
+	}
+	if ua.filter == nil {
+		return nil, newMissingArgumentError("filter")
+	}
+	if ua.update == nil {
+		return nil, newMissingArgumentError("update")
+	}
+
+	return ua, nil
+}
+
+type updateOneArguments struct {
+	filter bson.Raw
+	update interface{}
+	opts   *options.UpdateOneOptionsBuilder
+}
```

Comment:
You can avoid creating a new type here by separating `filter` and `update` from the options builder, allowing you to reuse `updateArguments` in both functions.

E.g. updated function signatures:
```go
func createUpdateOneArguments(args bson.Raw) (*updateArguments, *options.UpdateOneOptionsBuilder, error)
func createUpdateManyArguments(args bson.Raw) (*updateArguments, *options.UpdateManyOptionsBuilder, error)
```